AndroidPubSub/AuthenticatedPubsub.md

```diff
@@ -0,0 +1,85 @@
+## Using AWS IOT with user sign-in
+
+The sample demonstrates enabling guest users to use the AWS IoT APIs to securely publish-to and subscribe-from MQTT topics. Following steps demonstrates usage of AWS IoT with user sign-in. 
+
+1. Follow the steps [here](https://aws-amplify.github.io/docs/android/authentication) to add authentication UI to your app.
+
+2. Attach the following IAM Policy to the role of authenticated pool to authenticate and authorize the Cognito user to communicate with AWS IoT
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "iot:AttachPrincipalPolicy",
+                "iot:Connect",
+                "iot:Publish",
+                "iot:Subscribe",
+                "iot:Receive"
+            ],
+            "Resource": [
+                "*"
+            ]
+        }
+    ]
+}
+```
+
+3. Get the credentials provider
+
+```
+credentialsProvider =
+               IdentityManager.getDefaultIdentityManager().getCredentialsProvider();
+```
+
+4. In the app, after the user logs in using the sign-in UI, get CognitoIdentity from credentialsProvider and use it to attach principal policy to the IotAndroidClient. Create IoT policy as specified in step 4 of [Using the sample](https://github.com/awslabs/aws-sdk-android-samples/tree/master/AndroidPubSub#using-the-sample) section above.
+
+```
+// Setup the logins map
```

Comment:
Add more comments here to describe what each step is doing.
For example, "//Setup the logins map" doesn't tell you why we need the logins map.